apis/Google.Devtools.AspNet/Google.Devtools.AspNet.IntegrationTests/CloudTrace/CloudTraceIntregrationTest.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using Google.Cloud.Trace.V1;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+using Trace = Google.Cloud.Trace.V1.Trace;
+
+namespace Google.Devtools.AspNet.IntegrationTests
+{
+    public class CloudTraceIntregrationTest
+    {
+        /// <summary>Total time to spend sleeping when looking for a trace.</summary>
+        private const int TimeoutMilliseconds = 10000; // 10 seconds.
+
+        /// <summary>Time to sleep between checks for a trace.</summary>
+        private const int SleepIntervalMilliseconds = 1000; // 1 second.
+
+        /// <summary>Project id to run the test on.</summary>
+        private readonly string ProjectId;
+
+        /// <summary>Unique id of the test.</summary>
+        private readonly string TestId;
+
+        /// <summary>Test start time to allow for easier querying of traces.</summary>
+        private readonly Timestamp StartTime;
+
+        /// <summary>Using the underlying grpc client so we can filter.</summary>
+        private readonly TraceService.TraceServiceClient GrpcClient;
+
+        public CloudTraceIntregrationTest()
+        {
+            ProjectId = Utils.GetProjectIdFromEnvironment();
+            TestId = Utils.GetTestId();
+            StartTime = Timestamp.FromDateTime(DateTime.Now.ToUniversalTime());
+            GrpcClient = TraceServiceClient.Create().GrpcClient;
+        }
+
+        private GrpcTraceConsumer CreateGrpcTraceConsumer()
+        {
+            TraceServiceClient client = TraceServiceClient.Create();
+            return new GrpcTraceConsumer(Task.FromResult(client));
+        }
+
+        private SimpleManagedTracer CreateSimpleManagedTracer(ITraceConsumer consumer)
+        {
+            TraceIdFactory traceIdFactory = TraceIdFactory.Create();
+            Trace trace = new Trace()
```

Comment:
Nit: no need for the `()` at the end here are you're using an object initializer.